mtbl/threadpool.c

```diff
@@ -0,0 +1,382 @@
+/*
+ * Copyright (c) 2024 DomainTools LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <stdlib.h>
+#include <pthread.h>
+#include <assert.h>
+#include <stdbool.h>
+#include <mtbl.h>
+
+#include "threadpool.h"
+
+/* Public interface */
+
+struct mtbl_threadpool *
+mtbl_threadpool_init(size_t thread_count) {
+
+	struct mtbl_threadpool *pool = calloc(1, sizeof(*pool));
+
+	if (thread_count > 0)
+		pool->pool = threadpool_init(thread_count);
+
+	return pool;
+}
+
+void
+mtbl_threadpool_destroy(struct mtbl_threadpool **poolp) {
+
+	struct mtbl_threadpool *pool = *poolp;
+	if (pool == NULL)
+		return;
+	threadpool_destroy(&pool->pool);
+	free(pool);
+	*poolp = NULL;
+}
+
+
+/* Threadpool */
```

Comment:
a few more comments please on the structs.

Reply:
Pushed